src/instruction.rs

```diff
@@ -152,7 +154,24 @@ impl AddressingMode {
     }
 }
 
-pub type DecodedInstr = (Instruction, OpInput);
+pub struct DecodedInstr(pub Instruction, pub OpInput);
+
+impl Display for DecodedInstr {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
+        match self.1 {
+            OpInput::UseImplied => write!(f, "{:?}", self.0),
+            OpInput::UseImmediate(v) => write!(f, "{:?} #${:02X}", self.0, v),
+            OpInput::UseRelative(v) => write!(f, "{:?} ${:04X}", self.0, v),
+            OpInput::UseAddress(v) => write!(f, "{:?} ${:04X}", self.0, v),
```

Comment:
How does this code handle indexed and indirect addressing modes? I would expect format strings for things like `adc ($1234),x` or `adc ($12),y` for example.

Reply:
It doesn't at the moment. This just impls Display for `DecodedInstr`, which is defined as
```rust
pub struct DecodedInstr(pub Instruction, pub OpInput);
```

where `OpInput` is 

```rust
pub enum OpInput {
    UseImplied,
    UseImmediate(u8),
    UseRelative(u16),
    UseAddress(u16),
}
```

We don't have the addressing mode in there. 😢 Instead, there is 

```rust
pub static OPCODES: [Option<(Instruction, AddressingMode)>; 256] = [
    /*0x00*/
    Some((Instruction::BRK, AddressingMode::Implied)),
    /* ... */
```

which does the mapping to the `AddressingMode`. I've added that to the print output. Can you have another look to see if it's better now and covers all of our cases?